Alliance.Client/Extensions/VOIP/ViewModels/VoipVM.cs

```diff
@@ -64,10 +85,26 @@ public void OnTick(float dt)
         {
             for (int i = 0; i < ActiveVoicePlayers.Count; i++)
             {
-                if (!ActiveVoicePlayers[i].IsMyPeer && ActiveVoicePlayers[i].UpdatesSinceSilence >= 30)
+                if (ActiveVoicePlayers[i] != null && !ActiveVoicePlayers[i].IsMyPeer)
                 {
-                    ActiveVoicePlayers.RemoveAt(i);
-                    i--;
+                    ActiveVoicePlayers[i].UpdatesSinceSilence++;
+                    if (ActiveVoicePlayers[i].UpdatesSinceSilence >= 30)
+                    {
+                        ActiveVoicePlayers.RemoveAt(i);
+                        i--;
+                    }
```

Comment:
In order to handle time I would recommand to use `DateTime` and to define a property that will save the last time where  the player spoke and to check if  this time is bigger than a fixed value.

With that, you avoid updating a value each tick.

You can find an example in `SaeBehavior.cs` `line 347`

e.g. : 
```csharp
if (
  DateTime.Now.Ticks - lastTimeMakerPosed.Ticks >= 1500000 && 
  Input.IsKeyDown(createMarkerIK.KeyboardKey.InputKey)
)
            {
                lastTimeMakerPosed = DateTime.Now;
                SendToServerToCreateMarker();
            }
```

PS : a comment or a constant would be better than just putting "1500000" since this is hard to know how much it represent

Reply:
Agreed, this updating mechanic could be better. I'm letting it as is for now since MPVoicePlayerVM is a native component and was designed like this, but we can scrap it the day we change the VOIP interface.